src/dfcx_scrapi/tools/test_cases_util.py

```diff
@@ -0,0 +1,552 @@
+                return None
+
+        return target_test_config
+
+    def _get_flow_id_from_test_config(
+        self,
+        test_case: types.TestCase) -> str:
+        """Attempt to get the Flow ID from the Test Case Test Config."""
+
+        if "flow" in test_case.test_config:
+            return test_case.test_config.flow
+        elif "page" in test_case.test_config:
+            return "/".join(test_case.test_config.page.split("/")[:8])
+        else:
+            agent_id = "/".join(test_case.name.split("/")[:6])
+            return f"{agent_id}/flows/00000000-0000-0000-0000-000000000000"
+
+    def _get_page_id_from_test_config(
+        self,
+        test_case: types.TestCase,
+        flow_id: str) -> str:
+        """Attempt to get the Page ID from the Test Case Test Config."""
+
+        if "page" in test_case.test_config:
+            return test_case.test_config.page
+        else:
+            return f"{flow_id}/pages/START_PAGE"
+
+    def _get_flow_display_name(
+        self,
+        flow_id: str,
+        flows_map: Dict[str, str]) -> str:
+        """Attempt to get the Flow Display Name from the Flows Map
+            by the flow ID."""
+
+        flow = flows_map.get(flow_id, None)
+        return flow
+
+    def _get_page_display_name(
+        self,
+        flow_id: str,
+        page_id: str,
+        pages_map: Dict[str, Dict[str, str]]) -> str:
+        """Get the Page Display Name from the Pages Map based on the Page ID."""
+
+        page_map = pages_map.get(flow_id, None)
+        page = "START_PAGE"
+
+        if page_map:
+            page = page_map.get(page_id, None)
+
+        return page
+
+    def _get_flow_id_by_flow_name(
+        self,
+        flow_name: str,
+        flows_map: Dict[str, str]) -> str:
+        """Attempt to get the Flow id from the Flows Map
+            by the flow display name."""
+
+        flows_map = {name: id for id, name in flows_map.items()}
+        flow_id = flows_map.get(flow_name, None)
+
+        return flow_id
+
+    def _get_page_id_by_page_name(
+        self,
+        flow_id: str,
+        page_name: str,
+        pages_map: Dict[str, Dict[str, str]]) -> str:
+        """Attempt to get the Page id from the Pages Map
+            by the Page display name."""
+
+        if not flow_id:
+            return None
+
+        page_map = pages_map.get(flow_id, None)
+        page_id = None
+        if  page_name in [
+            "Start Page",
+            "End Flow",
+            "End Session",
+            "End Flow With Failure",
+            "End Flow With Human Escalation",
+            "End Flow With Cancellation"
+            ]:
+            page_name = page_name.upper().replace(" ", "_")
+            page_id = f"{flow_id}/pages/{page_name}"
+        elif page_map:
+            page_map = {name: id for id, name in page_map.items()}
+            page_id = page_map.get(page_name, None)
+
+        return page_id
+
+    def _get_intent_id_by_intent_name(
```

Comment:
Yes, I agree... I think it would be efficient if I store both intents_map, pages_map, flows_map (standard and reverse) in common_config dict. So, the method calls them once if necessary. 

--Flows map ---
`    def _get_flow_id_by_flow_name(
        self,
        flow_name: str,
        flows_map: Dict[str, str]) -> str:
        """Attempt to get the Flow id from the Flows Map
            by the flow display name."""

        **flows_map = {name: id for id, name in flows_map.items()}**
        flow_id = flows_map.get(flow_name, None)`

--Pages map---
`elif page_map:
            **page_map = {name: id for id, name in page_map.items()}**
            page_id = page_map.get(page_name, None)`

Reply:
`    def _get_commons_config(
        self,
        agent_id: str) -> Dict[str, Dict[str, str]]:
        """This method map flows, pages, and intents
            are in the dictionary format
          Arg:
            agent_id: The formatted CX Agent ID
          Returns:
            a dictionary of flows, pages, and intents
        """
        commons_config = {}
        dfcx_flows = flows.Flows(creds=self.creds, agent_id=agent_id)
        dfcx_pages = pages.Pages(creds=self.creds)
        dfcx_intents = intents.Intents(creds=self.creds, agent_id=agent_id)
        dfcx_intents_map = dfcx_intents.get_intents_map(agent_id=agent_id)
        dfcx_flows_map = dfcx_flows.get_flows_map(agent_id=agent_id)
        dfcx_pages_map = {}
        for flow_id in dfcx_flows_map.keys():
            dfcx_pages_map[flow_id] = (
                dfcx_pages.get_pages_map(
                  flow_id=flow_id
                )
            )
        commons_config["flows"] = dfcx_flows
        commons_config["pages"] = dfcx_pages
        commons_config["intents"] = dfcx_intents
        commons_config["flows_map"] = dfcx_flows_map
        commons_config["pages_map"] = dfcx_pages_map
        commons_config["intents_map"] = dfcx_intents_map

        commons_config["flows_map_reverse"] = (
            self._reverse_dict(commons_config["flows_map"])
        )
        commons_config["intents_map_reverse"] = (
            self._reverse_dict(commons_config["intents_map"])
        )
        commons_config["pages_map_reverse"] = (
            {
                f_id:self._reverse_dict(pages_map) for f_id, pages_map in dfcx_pages_map.items()
            }
        )

        return commons_config`